modules/azure_aks/main.tf

```diff
@@ -13,7 +13,7 @@ resource "azurerm_kubernetes_cluster" "aks" {
   kubernetes_version                 = var.kubernetes_version
   api_server_authorized_ip_ranges    = var.aks_cluster_endpoint_public_access_cidrs
   private_cluster_enabled            = var.aks_private_cluster
-  private_dns_zone_id                = var.aks_private_cluster ? "System" : null
+  private_dns_zone_id                = var.aks_private_cluster && var.aks_private_cluster_private_dns_zone_value != "" ? var.aks_private_cluster_private_dns_zone_value : (var.aks_private_cluster ? "System" : null)
```

Comment:
The variable name will need to be changed to `aks_cluster_private_dns_zone_id`

When adjusting for public and private you’ll also need to figure out if you need to use `dns_prefix` for public facing clusters or `dns_prefix_private_cluster` when using private facing clusters. I would imagine this is where the error may be coming from. You’ll need to read through the entire resource document found here: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#argument-reference

Reply:
I made a comparison to either define `dns_prefix` or `dns_prefix_private_cluster ` depending on the cluster being private or not. Should have `dns_prefix_private_cluster ` been defined prior to this PR as the code already gave the possibility of creating a private aks cluster? Thank you

Reply:
After trying this whole PR implementation, I think that the previous message was the case as everything seems to be working as expected.